polaris/evaluate/_metric.py

```diff
@@ -84,55 +164,241 @@ class Metric(Enum):
     roc_auc_ovo = MetricInfo(
         fn=roc_auc_score, kwargs={"multi_class": "ovo"}, direction="max", y_type="y_score"
     )
-    # TODO: add metrics to handle multitask multiclass predictions.
 
     # docking related metrics
     rmsd_coverage = MetricInfo(fn=rmsd_coverage, direction="max", y_type="y_pred")
 
+
+class Metric(BaseModel):
+    """
+    A Metric in Polaris.
+
+    Any metric implementation in Polaris object wraps a BaseMetric. This class holds the most vanilla implementation.
+    It does some minor preprocessing on the inputs and then simply passes these arguments to the BaseMetric.
+
+    It serves as a base class for more complex metrics.
+
+    Attributes:
+        metric: The actual callable that is at the core of the metric implementation.
+        kind: The kind of metric, uses to discriminate different types of metrics in a discriminated union.
+        name: The name of the metric. The metrics for a specific benchmark need to be uniquely named.
+            In most cases, a sensible default will be set and the user doesn't need to worry about this.
+        config: For more complex metrics, this object should hold all parameters for the metric.
+    """
+
+    metric: BaseMetric
+    kind: Literal["default"] = "default"
+    name: str | None = None
+    config: BaseModel | None = None
```

Comment:
**Question**
I think this is where we should be using the discriminated union. For now, this should be:
```python
config: GroupedMetricConfig | None = None
```

Then the `kind` can actually be derived from what config model is defined. If there's nothing, then it's a default metric. If there a `GroupedMetricConfig`, then it's a grouped metric. If we have new types of metric in the future, then new configuration models will define what they are.